routing/single_vehicle_world_graph.hpp

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#include "routing/cross_mwm_graph.hpp"
+#include "routing/edge_estimator.hpp"
+#include "routing/geometry.hpp"
+#include "routing/index_graph.hpp"
+#include "routing/index_graph_loader.hpp"
+#include "routing/num_mwm_id.hpp"
+#include "routing/road_graph.hpp"
+#include "routing/segment.hpp"
+#include "routing/world_graph.hpp"
+
+#include "geometry/point2d.hpp"
+
+#include <memory>
+#include <vector>
+
+namespace routing
+{
+class SingleVehicleWorldGraph final : public WorldGraph
+{
+public:
+  SingleVehicleWorldGraph(std::unique_ptr<CrossMwmGraph> crossMwmGraph,
+                          std::unique_ptr<IndexGraphLoader> loader,
+                          std::shared_ptr<EdgeEstimator> estimator);
+
+  // |isEnding| == true iff |segment| is first or last segment of the route. Needed because first
+  // and last segments may need special processing.
+  void GetEdgeList(Segment const & segment, bool isOutgoing, bool isLeap, bool isEnding,
+                   std::vector<SegmentEdge> & edges) override;
+
+  Junction const & GetJunction(Segment const & segment, bool front) override;
+  m2::PointD const & GetPoint(Segment const & segment, bool front) override;
+  RoadGeometry const & GetRoadGeometry(NumMwmId mwmId, uint32_t featureId) override;
+
+  // Clear memory used by loaded index graphs.
```

Comment:
Do we need comments to overrides? From my point of view, no, because you just repeat the comment from the base class.

Reply:
+1 for not commenting overrides if they are commented in base.

Reply:
removed